client/components/date-range/index.js

```diff
@@ -93,10 +94,12 @@ export class DateRange extends Component {
 		// Build initial state
 		this.state = {
 			popoverVisible: false,
-			staleStartDate: null,
-			staleEndDate: null,
+			staleStartDate: startDate,
+			staleEndDate: endDate,
```

Comment:
I set staleDates initially because when you open the dates and the click cancel without changing anything it empty the dates field and display the date format instead. 

This is because even when cancel is clicked it set state with staleDate which would be null if anything wasn't set.

```js
revertDates = () => {
		this.setState(
			( previousState ) => {
				const startDate = previousState.staleStartDate;
				const endDate = previousState.staleEndDate;
				const newState = {
					staleDatesSaved: false,
					startDate: startDate,
					endDate: endDate,
					textInputStartDate: this.toDateString( startDate ),
					textInputEndDate: this.toDateString( endDate ),
				};

				return newState;
			},
			() => {
				this.props.onDateCommit( this.state.startDate, this.state.endDate );
			}
		);
	};
```

![image](https://user-images.githubusercontent.com/47489215/228636241-680a97f4-5591-48bc-bfc5-2d227ce1f6ba.png)